docs/sources/userguide/plugins.md

```diff
@@ -0,0 +1,157 @@
+Docker Plugins MVP
+==================
+
+Plugins provide a mechanism for hooking into the Docker engine to extend its behavior.
+
+HTTP over a local UNIX socket is used to communicate between Docker and its plugins.
+
+Plugins are distributed as containers.
```

Comment:
We should definitely not bind-mount the docker socket into the plugins.
Once people start writing plugins which assume this, it will be impossible
to change the design without breaking them.

On Thu, Mar 12, 2015 at 4:20 AM, lukemarsden <notifications@github.com>
wrote:

> In docs/sources/userguide/plugins.md
> <https://github.com/cpuguy83/docker/pull/3#discussion_r26293764>:
>
> > @@ -0,0 +1,157 @@
> > +Docker Plugins MVP
> > +==================
> > +
> > +Plugins provide a mechanism for hooking into the Docker engine to extend its behavior.
> > +
> > +HTTP over a local UNIX socket is used to communicate between Docker and its plugins.
> > +
> > +Plugins are distributed as containers.
>
> Agree. Caveat here is that if we bind-mount the docker socket, then any
> plugin can elevate its own privileges. This needs more thought but not yet
> for an extensions MVP.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cpuguy83/docker/pull/3/files#r26293764>.
>